x-pack/heartbeat/monitors/browser/browser.go

```diff
@@ -35,12 +35,10 @@ func create(name string, cfg *common.Config) (p plugin.Plugin, err error) {
 		logp.Info("Synthetic browser monitor detected! Please note synthetic monitors are a beta feature!")
 	})
 
-	curUser, err := user.Current()
-	if err != nil {
-		return plugin.Plugin{}, fmt.Errorf("could not determine current user for script monitor %w: ", err)
-	}
-	if curUser.Uid == "0" {
-		return plugin.Plugin{}, fmt.Errorf("script monitors cannot be run as root! Current UID is %s", curUser.Uid)
+	// We do not use user.Current() which does not reflect setuid changes!
+	if syscall.Geteuid() == 0 {
+		euid := syscall.Geteuid()
```

Comment:
isn't this basically a dead code as its 0?

Reply:
Good catch lol